account_bank_statement_clearing_account/models/account_bank_statement.py

```diff
@@ -0,0 +1,82 @@
+# coding: utf-8
+from odoo import api, models
+
+
+class BankStatement(models.Model):
+    _inherit = 'account.bank.statement'
+
+    @api.multi
+    def get_reconcile_clearing_account_lines(self):
+        if (self.journal_id.default_debit_account_id !=
+                self.journal_id.default_credit_account_id or
+                not self.journal_id.default_debit_account_id.reconcile):
+            return False
+        account = self.journal_id.default_debit_account_id
+        currency = self.journal_id.currency_id or self.company_id.currency_id
+
+        def get_bank_line(st_line):
+            for line in st_line.journal_entry_ids:
+                if st_line.amount > 0:
+                    compare_amount = st_line.amount
+                    field = 'debit'
+                else:
+                    compare_amount = -st_line.amount
+                    field = 'credit'
+                if (line[field] and
+                        not currency.compare_amounts(
+                            line[field], compare_amount) and
+                        line.account_id == account):
+                    return line
+            return False
+
+        move_lines = self.env['account.move.line']
+        for st_line in self.line_ids:
+            bank_line = get_bank_line(st_line)
+            if not bank_line:
+                return False
+            move_lines += bank_line
+        balance = sum(line.debit - line.credit for line in move_lines)
+        if not currency.is_zero(balance):
+            return False
+        return move_lines
+
+    @api.multi
+    def reconcile_clearing_account(self):
+        self.ensure_one()
+        lines = self.get_reconcile_clearing_account_lines()
+        if not lines:
+            return False
+        if any(line.full_reconcile_id or
+               line.matched_debit_ids or
+               line.matched_credit_ids
+               for line in lines):
+            return False
+        lines.reconcile()
```

Comment:
this was lines.partial_reconcile. I think this replaces that functionality. Not sure though.
Instead of having all lines fully reconciled I get extra partial reconciled lines. Because of this I check this case in test here https://github.com/VanMoof/bank-statement-import/pull/1/files#diff-cb3a74c3e8b6c20ec292dd7246ce65bdR44
Now it fails because of those extra partial reconciled lines.

Reply:
UPDATE:
I replaced lines.reconcile() above with lines.check_full_reconcile() and now the test passes. Even though it passed I suspect it is wrong. We can use here reconcile(), check_full_reconcile() and auto_reconcile_lines(). Only the check_full_reconcile() creates one full reconcile record with all of the lines. The other 2 create one full reconciliation record and partial records for each line in the reconciliation. 
I debugged all of the methods and did not get any wiser. My hunch is that we should use auto_reconcile_lines(). This would mean changing the test. I don't even get the business side of this kind of reconciliation where we reconcile the whole bank statement with it self. Instead of having for each line a match between bank statement account move line and invoice receivable account move line.